midx.c

```diff
@@ -1040,15 +1062,32 @@ int verify_midx_file(const char *object_dir)
 	}
 
 	progress = start_progress(_("Verifying object offsets"), m->num_objects);
+
+	/*
+	 * Create an array mapping each object to its packfile id.  Sort it
```

Comment:
Makes sense, and follows a pattern that exists elsewhere in this file (ie midx_repack()).  Clearly results in a huge perf win by avoiding opening/closing pack files all the time.